### PR TITLE
Adding message and chat HTML

### LINKED ISSUE #99: Create message and chat HTML
## Expected Behavior

as described in our project, we want to enable our users to talk to each other via chat

## Current Behavior

our project still missing this part

## Acceptance Criteria

Add HTML for both messages and chat

### PR DESCRIPTION
**Pull request by Yuval-Vino**
### Desired Outcome

Adding chat feature to Hotails

### Implemented Changes

Added message and chat html
updated message views in order to fit the HTML files
updated navbar chat button to redirect to message HTML
updated hotails urls added message and chat htmls
updated function to daycare model - get_daycare_primary_image_url (as implemnted in #89  for showing daycare picture in chat)
also added CSS for the relevant htmls

### Connected Issue/Story

Resolves #99 

#### Changelog

- [ ] The CHANGELOG has been updated
- [X] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage
 Added test for the HTML's that checking that we are in the right site
***test_shown_only_relevant_contacts***
***test_shown_only_relevant_message_in_chat***
***test_send_new_message***
***test_cant_send_blank_message***

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [X] This PR does not require updating any documentation

### before picking specifc user to talk
07174943-e82d-4df8-b131-7e900eb3ca22

### after choosing chat (as a dogowner)
58c7d26e-be18-4d03-9df8-bd2ae6b0eef2

### after choosing chat (as daycare)
21730a1c-eaa7-47c9-beb3-4193a8b29a44

### CODE REVIEW

**Review by ErezCohenn — COMMENTED** [review]
Niceee @Yuval-Vino 
don't forget to assign yourself, and add `ready for review` label,
and the most important tests are missing

**Comment by Yuval-Vino** [comment]
> Niceee @Yuval-Vino don't forget to assign yourself, and add `ready for review` label, and the most important tests are missing

Done! 
Added test
### this PR is ready for reviews!!

File: daycare/models.py [hunk 1]
```diff
@@ -50,6 +50,12 @@ def create(email, username, password, name, description, price_per_day, capacity
 
         return new_daycare
 
+    def get_daycare_primary_image_url(self):
+        daycare_images = Image.get_images_by_daycare_id(daycare_id=self.id)
+        if daycare_images is not None and daycare_images.first() is not None:
+            return daycare_images.first().url
+        return "../../static/images/daycare-default-profile-image.jpeg"
+
```
Comment (ErezCohenn) [inline daycare/models.py:59]: This code was implemented in #89 right? so I think it needs to be in another separate commit and add in the description on the PR that this PR depends on #89

File: message/test_html.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+from daycare.models import DayCare
+import pytest
+import random
+
+
+@pytest.mark.django_db()
+class TestMessageHtml:
+    def test_shown_only_relevant_contacts(self, client):
+        daycare = random.choice([daycare for daycare in DayCare.objects.all() if 'static' in daycare.user.username])
+        client.force_login(user=daycare.user)
```
Comment (ErezCohenn) [inline message/test_html.py:11]: overall in tests, I think you cant use random methods, so if the test fails or succeeds you cant detect what made the reason for the tests to succeed/fail, therefor its should be explicit.

**Review by ErezCohenn — CHANGES_REQUESTED** [review]
left some comments below, @Yuval-Vino take a look and tell me what you think

File: message/templates/messages.html [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                              </div>
+                      {% endif %}
+
+                  </div>
+                </div>
+                </a>
+            {% endfor %}
+
+                    </tbody>
+                </table>
```
Comment (Omeramsc) [inline message/templates/messages.html:42]: Is there a table here? where is the opening tag?

File: message/templates/messages.html [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                                <h5>{{ contact.name }}</h5>
+                              </div>
+                      {% endif %}
+
+                  </div>
+                </div>
+                </a>
+            {% endfor %}
+
+                    </tbody>
```
Comment (Omeramsc) [inline message/templates/messages.html:41]: I don't see a <tbody> tag.

**Comment by Omeramsc** [comment]
Please lint and re-orgenize `message/templates/messages.html`, there is a mess in the tags there (closing tags with no opening tags, and seems like divs that are not closed either? hard to read)

File: message/test_html.py (hunk 2 above)
Comment (Omeramsc) [reply message/test_html.py:11]: Tests should never have a random element in them. you should test all the **different** relevant possibilities explicitly. you should be able to depend on the test results.

File: daycare/models.py (hunk 1 above)
Comment (Yuval-Vino) [reply daycare/models.py:59]: It is indicated in the description

File: message/test_html.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        response = client.get('/messages/')
+        assert response.status_code == 200
```
Comment (Omeramsc) [inline message/test_html.py:13]: we already mention the status code tests are redundant...

File: message/templates/messages.html (hunk 3 above)
Comment (Yuval-Vino) [reply message/templates/messages.html]: you're right, this is leftovers from a different try. 
Removed!

File: message/templates/messages.html (hunk 4 above)
Comment (Yuval-Vino) [reply message/templates/messages.html]: you're right, this is leftovers from a different try. 
Removed!

File: message/test_html.py (hunk 2 above)
Comment (Yuval-Vino) [reply message/test_html.py]: Thanks for the good reviews guys!
removed random, now we test all users and all their contacts

**Comment by Yuval-Vino** [comment]
> left some comments below, @Yuval-Vino take a look and tell me what you think

thanks for the reviews, I address them all :)

**Comment by Yuval-Vino** [comment]
> Please lint and re-orgenize `message/templates/messages.html`, there is a mess in the tags there (closing tags with no opening tags, and seems like divs that are not closed either? hard to read)

Done! also addressed all of your comments , let me know if you see more possible changes to improve

**Review by ErezCohenn — APPROVED** [review]
LGTM !
don't forget to rebase.
also I think you should squash the commit `Fixing the PR after comments` to `Adding message and chat HTML`

**Review by OfirMatasas — APPROVED** [review]
Looks amazing! great job, @Yuval-Vino

**Review by tamirmatok — APPROVED** [review]
LGTM ! great job @Yuval-Vino .

**Review by ormergi — CHANGES_REQUESTED** [review]
Hi Yuval, please squash intermediate commits

**Comment by Yuval-Vino** [comment]
> Hi Yuval, please squash intermediate commits

Done

File: main/templates/main/base_template.html [hunk 6]
```diff
@@ -27,7 +27,7 @@
                     <li class="nav-item"><a class="nav-link text-white" href="/profile">Profile</a></li>
                     <li class="nav-item"><a class="nav-link text-white" href="">Orders</a></li>
                     <li class="nav-item"><a class="nav-link text-white" href="">Search</a></li>
-                    <li class="nav-item"><a class="nav-link text-white" href="">Chats</a></li>
+                    <li class="nav-item"><a class="nav-link text-white" href="/messages">Chats</a></li>
```
Comment (ormergi) [inline main/templates/main/base_template.html:30]: nit: I would call it chat

**Review by ormergi — CHANGES_REQUESTED** [review]
Hi Yuval, 
Please fix the commit message and body.
Also see my inline comments.

File: message/test_html.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
+from .models import Message
+from daycare.models import DayCare
+import pytest
+
+
+@pytest.mark.django_db()
+class TestMessageHtml:
```
[the remark this answers is not in the document]
Comment (Yuval-Vino) [reply message/test_html.py]: Done

File: message/test_html.py [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    def test_shown_only_relevant_contacts(self, client):
+        static_daycares = [daycare for daycare in DayCare.objects.all() if 'static' in daycare.user.username]
+        for daycare in static_daycares:
+            client.force_login(user=daycare.user)
+            response = client.get('/messages/')
+            assert response.status_code == 200
```
[the remark this answers is not in the document]
Comment (Yuval-Vino) [reply message/test_html.py:14]: Done

File: main/templates/main/base_template.html (hunk 6 above)
Comment (Yuval-Vino) [reply main/templates/main/base_template.html:30]: name chat is occupied for chat between 2 users.
From '/messages' you can be redirected to '/chat' with a specific chat partner

**Comment by Omeramsc** [comment]
Please resolve conflicts ASAP if you want it in for the demo.
There is room for improvements, but It is good enough to get in.

**Comment by Omeramsc** [comment]
+1 LGTM
